File: problems/006/006.go
package main

import "fmt"
import "os"
import "strconv"
import "time"

func f() uint64 {
    sum, sumOfSquares := uint64(0), uint64(0)

    for i := uint64(1); i < 101; i++ {
        sum += i
        sumOfSquares += i * i
    }

    squaredSum := sum * sum

    return squaredSum - sumOfSquares
}

func main() {
    if len(os.Args) == 1 {
        fmt.Println(f())
    } else if len(os.Args) == 2 {
        iters, _ := strconv.ParseInt(os.Args[1], 10, 64)

        start := time.Now()
        for i := int64(0); i < iters; i++ {
            f()
        }
        end := time.Now()

        fmt.Println(end.Sub(start).Nanoseconds())
    }
}

File: problems/006/006.rs
extern crate test;
extern crate time;

use std::iter::AdditiveIterator;
use std::os::args;
use test::black_box;
use time::precise_time_ns;

#[inline]
fn f() -> u64 {
    let sumOfSquares = range(1u64, 101).map(|x| x * x).sum();
    let sum = range(1u64, 101).sum();
    let squaredSum = sum * sum;

    return squaredSum - sumOfSquares;
}

fn main() {
    match args().as_slice() {
        [_] => {
            println!("{}", f());
        },
        [_, ref iters] => {
            let iters: u64 = from_str(iters.as_slice()).unwrap();

            let start = precise_time_ns();
            for _ in range(0, iters) {
                black_box(f());
            }
            let end = precise_time_ns();

            println!("{}", end - start);
        },
        _ => unreachable!(),
    }
}
